src/banner/Banner.jsx

```diff
@@ -0,0 +1,96 @@
+import { BANNER_TYPES } from './private/utils';
+
+/* eslint-disable react/no-unused-prop-types */
+const propTypes = {
+  /**
+   * An action element to be added to the action section of the banner.
+   */
+  action: PropTypes.element,
+  /**
+   * The pieces to populate a banner when type=`custom`.
+   */
+  custom: PropTypes.shape({
+    /**
+     * The title for the banner which will be bolded.
+     */
+    bannerTitle: PropTypes.string,
+    /**
+     * The class name used to set the icon in the banner.
+     */
+    icon: PropTypes.element,
+    /**
+     * Sets an author-defined class, to control the status bar color to be used.
+     */
+    colorClass: PropTypes.string,
+  }),
+  /**
+   * Nodes providing the message content for the banner. Can contain text and HTML.
+   */
+  description: PropTypes.node,
+  /**
+   * Callback function triggered when Dismiss button is clicked. The presence of this prop will cause
+   * the Dismiss button to be included on the banner.
+   */
+  onDismiss: PropTypes.func,
+  /**
+   * The type of alert to be rendered. One of `alert`, `error`, `warning`, `info`, `success`, `unsatisfied`, `unverified` or `custom`.
+   */
+  type: PropTypes.oneOf([
+    BANNER_TYPES.ALERT,
+    BANNER_TYPES.ERROR,
+    BANNER_TYPES.WARNING,
+    BANNER_TYPES.UNSATISFIED,
+    BANNER_TYPES.UNVERIFIED,
+    BANNER_TYPES.ADVISORY,
+    BANNER_TYPES.INFO,
+    BANNER_TYPES.SUCCESS,
+    BANNER_TYPES.CUSTOM,
+  ]).isRequired,
+};
+/* eslint-enable react/no-unused-prop-types */
+
+const Banner = (props) => {
+  const bannerRegistration = React.useContext(BannerRegistrationContext);
+
+  React.useEffect(() => {
+    /**
+     * A unique identifier is generated for each Banner during construction. This will be used to
+     * uniquely register/unregister the banner with ancestor Banner Managers without requiring consumers to
+     * define unique identifiers themselves.
+     */
+    const uuid = uuidv4();
+
+    /**
+     * If the bannerRegistration value is the ProviderRegistrationContext's default value,
+     * then there is not a matching BannerCheckpoint above it in the hierarchy.
+     * This is possible but likely not intentional, so the component warns.
+     */
+    if (!bannerRegistration && process.env.NODE_ENV !== 'production') {
+      /* eslint-disable no-console */
+      console.warn('A Banner was not rendered within the context of a BannerCheckpoint. If this is unexpected, validate that the expected version of the terra-application package is installed.');
+      /* eslint-enable no-console */
+      return () => {};
+    }
+
+    if (bannerRegistration && bannerRegistration.registerBanner) {
+      bannerRegistration.registerBanner(uuid, props);
+    }
+
+    return () => {
+      if (bannerRegistration && bannerRegistration.unregisterBanner) {
+        bannerRegistration.unregisterBanner(uuid, props.type);
+      }
+    };
+  }, [bannerRegistration, props]);
```

Comment:
I _think_ the props object will always be different if this component is updated due to a parent render. You might want to spread the props out. Can you verify?

Reply:
You are right. I will need to re-think my approach on this. Anonymous  onDismiss functions, action and customIcons are all triggering re-renders after I flattened extracted & flattened down the prop types.

Reply:
updated: https://github.com/cerner/terra-application/pull/45/commits/a905caaef9296be5ffa5857c8281de69a4c9595f#diff-dc407e6487b82f8a1ebe8f58e5088673R96